hasher-matcher-actioner/webapp/src/components/settings/ActionPerformer/WebhookActioner.tsx

```diff
@@ -14,45 +14,45 @@ interface WebhookTypeInterface extends Record<string, any> {
   WebhookPutActionPerformer: string;
 }
 
+const ActionerTypes = {
+  WebhookActioner: {
+    args: {
+      url: {description: 'The url to send a webhook to'},
+      webhookType: {
+        description: 'What type of webhook should be sent?',
+        default: 'POST',
+      },
+      headers: {
+        description: 'Optional json object of headers to include in webhook',
+        default: '{}',
+      },
+    },
+    description:
+      'When a match occurs, a webhook will be sent to the specified url with data describing the match',
+  },
+};
+const WebhookType: WebhookTypeInterface = {
+  WebhookPostActionPerformer: 'POST',
+  WebhookGetActionPerformer: 'GET',
+  WebhookDeleteActionPerformer: 'DELETE',
+  WebhookPutActionPerformer: 'PUT',
+};
+const actionerDetails = ActionerTypes.WebhookActioner;
+
 type WebhookActioner = {
-  url: string;
-  headers: string;
-  webhookType: string;
+  action: Action;
   editing: boolean;
-  onChange: (key: string, keyValueMap: {[key: string]: string}) => void;
+  updateAction: (action: Action) => void;
 };
 
 export default function WebhookActioner({
-  url,
-  headers,
-  webhookType,
+  action,
   editing,
-  onChange,
+  updateAction,
 }: WebhookActioner): JSX.Element {
-  const ActionerTypes = {
-    WebhookActioner: {
-      args: {
-        url: {description: 'The url to send a webhook to'},
-        webhookType: {
-          description: 'What type of webhook should be sent?',
-          default: 'POST',
-        },
-        headers: {
-          description: 'Optional json object of headers to include in webhook',
-          default: '{}',
-        },
-      },
-      description:
-        'When a match occurs, a webhook will be sent to the specified url with data describing the match',
-    },
-  };
-  const WebhookType: WebhookTypeInterface = {
-    WebhookPostActionPerformer: 'POST',
-    WebhookGetActionPerformer: 'GET',
-    WebhookDeleteActionPerformer: 'DELETE',
-    WebhookPutActionPerformer: 'PUT',
-  };
-  const actionerDetails = ActionerTypes.WebhookActioner;
+  const [, setURL] = useState(action.params.url);
```

Comment:
nit: what does this state and the corresponding calls do?

Reply:
True. I don't think we need this. Will remove and retest

Reply:
Updated state